include/deal.II/base/symmetric_tensor.h

```diff
@@ -511,8 +511,14 @@ namespace internal
 /**
  * Provide a class that stores symmetric tensors of rank 2,4,... efficiently,
  * i.e. only store those off-diagonal elements of the full tensor that are not
- * redundant. For example, for symmetric 2x2 tensors, this would be the
+ * redundant. For example, for symmetric $2\times 2$ tensors, this would be the
  * elements 11, 22, and 12, while the element 21 is equal to the 12 element.
+ * Within this documentation, second order symmetric tensors are denoted as
+ * bold-faced upper-case Latin letters such as $\mathbf A, \mathbf B, \dots$
+ * or bold-faced Greek letters such as $\boldsymbol{\varepsilon}$,
+ * $\boldsymbol{\sigma}$. The cartesian coordinates of a second-order tensor
+ * such as $\mathbf A$ are represented as $A_{ij}$ where $i,j$ are indices
```

Comment:
I think it is important that we clearly distinguish a tensor quantity such as $mathbf A$ from a scalar quantity such as the element $A_{ij}$, especially when we use them in the same equation e.g.

![pic](https://latex.codecogs.com/svg.latex?\LARGE&space;\mathbf{A}&space;=&space;A_{ij}&space;\mathbf{e}_i&space;\otimes&space;\mathbf{e}_j)

However, I understand the inconsistency I have here for the rank-4 tensors, and I'll fix it. For 4th order tensors I will keep $\mathbb A$ but change its components to $\mathcal{A}__{ijkl}$. I don't support $A_{ijkl}$ because in some situations we might have a rank 2 tensor $\mathbf C$ and a rank 4 tensor $\mathbb C$ in the same equation and we need to distinguish them clearly if we write the equation in index notation (although the situation is very rare).

The rendered notations are as follows for $\mathbf A, A_{ij}, \mathbb{A}, A_{ijkl}, \mathcal{A}_{ijkl}$:

![here](https://latex.codecogs.com/svg.latex?\LARGE&space;\mathbf&space;A&space;\quad&space;A_{ij}&space;\quad&space;\mathbb{A}&space;\quad&space;A_{ijkl}&space;\quad&space;\mathcal{A}_{ijkl})